src/common/playlists/DynamicBanner.jsx

```diff
@@ -3,41 +3,55 @@ import thumbPlay from 'images/thumb-play.png';
 import { Link } from 'react-router-dom';
 import { MdArrowRightAlt } from 'react-icons/md';
 import { loadCoverImage } from 'common/utils/coverImageUtil';
+import Loader from 'common/spinner/spinner';
 
 const DynamicBanner = ({ randomPlay }) => {
   const [coverImage, setCoverImage] = useState(null);
-  const [loading, setLoading] = useState(false);
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
-    setLoading(true);
-
     async function setPlayCover() {
-      if (loading && randomPlay && randomPlay.cover) {
-        setCoverImage(randomPlay.cover);
-        setLoading(false);
-      } else {
-        setLoading(true);
-        // if it is not passed as a meta data
-        // check in the play folder for a cover image with the name cover
-        const coverImage = await loadCoverImage(randomPlay.slug);
-
-        if (coverImage) {
-          setCoverImage(coverImage);
-          setLoading(false);
+      try {
+        if (loading && randomPlay && randomPlay.cover) {
+          setCoverImage(randomPlay.cover);
         } else {
-          setCoverImage(thumbPlay);
+          // if it is not passed as a meta data
+          // check in the play folder for a cover image with the name cover
+          const coverImage = await loadCoverImage(randomPlay.slug);
 
-          console.error(
-            `Cover image not found for the play ${randomPlay.name}. Setting the default cover image...`
-          );
+          if (coverImage) {
+            setCoverImage(coverImage);
+          } else {
+            setCoverImage(thumbPlay);
+
+            console.error(
```

Comment:
Rather than logging the error on the console, we can prompt errors with `alert` or display a toast. this would be a better user experience.

Reply:
Yeah, that would be better.